### ISSUE TITLE
Alertmanager docker container sending "firing" Slack message but not "resolved" Slack message

### ISSUE THREAD

**Issue by benw10-1** [prometheus/alertmanager#3432]
**What did you do?**

Ran `alertmanager` inside docker container with many alerts with Prometheus data-source.

**What did you expect to see?**

Alert firing and sending Slack message in specified channel. Then once resolved, sending the `resolved` message.

**What did you see instead? Under which circumstances?**

Alert firing and sending Slack message in specified channel. Then once resolved, it did not send the `resolved` message.

**Environment**

* System information:

	`Linux 4.14.318-241.531.amzn2.x86_64 x86_64`

* Alertmanager version:

```
docker exec -it 658d857649ad alertmanager --version
alertmanager, version 0.16.1 (branch: HEAD, revision: 571caec278be1f0dbadfdf5effd0bbea16562cfc)
  build user:       root@3000aa3a06c5
  build date:       20190131-15:05:40
  go version:       go1.11.5
```

* Prometheus version:

```
docker exec -it b356a588e819 prometheus --version
prometheus, version 2.1.0 (branch: HEAD, revision: 85f23d82a045d103ea7f3c89a91fba4a93e6367a)
  build user:       root@6e784304d3ff
  build date:       20180119-12:01:23
  go version:       go1.9.2
```

* Docker compose entry:
```
alertmanager:
    image: prom/alertmanager
    ports:
      - 9093:9093
    user: 1000:1000
    volumes:
      - (omitted)
    command:
      - '--config.file=(below config)'
      - '--storage.path=(omitted, but is volume path)'
      - '--web.external-url=(omitted)'
      - '--web.route-prefix=/'	
      - '--log.level=debug'
    restart: unless-stopped
```

* Alertmanager configuration file:
Will not include whole config, relevant config only is included (comment for more aspects if necessary)
```
global:
  scrape_interval:     15s # By default, scrape targets every 15 seconds.
  evaluation_interval: 15s # By default, scrape targets every 15 seconds.

  external_labels:
    monitor: (omitted)

# Load and evaluate rules in this file every 'evaluation_interval' seconds.
rule_files:
  - 'alert-generated.rules'

# alert
alerting:
  alertmanagers:
  - scheme: http
    static_configs:
    - targets:
      - "alertmanager:9093"

scrape_configs:
  - job_name: (omitted)
    static_config:
      - targets: ['(target omitted)']

route:
  receiver: 'slack'
	group_by: ['alertname']

receivers:
  - name: 'slack'
    slack_configs:
      - send_resolved: true
        channel: '#alerts'
	api_url: '(omitted)'
	text: "{{ .CommonAnnotations.summary }}"
```
`alert-generated.rules` (omitted many other alerts, but this is the only one causing the issue)
```
groups:
- name: generated
	rules:
	  - alert: proxy server down
	     expr: proxy_server_status{}  != 1
	     for: 3m
	     labels:
	       (omitted)
	     annotations:
                summary: (omitted)
```

* Prometheus configuration file:
(not relevant)

* Logs:
```
level=debug ts=2023-07-31T03:46:40.685571638Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:46:55.708689962Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:47:10.686200498Z caller=dispatch.go:343 component=dispatcher aggrGroup="{}:{alertname=\"proxy server down\"}" msg=flushing alerts="[proxy server down[fc7e0b6][active]]"
level=debug ts=2023-07-31T03:47:10.702097733Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:47:25.694143307Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:47:40.726868187Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:47:55.710934901Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:48:10.695793375Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:48:25.721533305Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:48:40.679542139Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:48:55.716326234Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:49:10.687724426Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:49:25.705784282Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:49:40.704911659Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:49:55.674943239Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:50:10.679015294Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:50:25.684075897Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:50:40.710773718Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:50:55.707091369Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:51:10.702619015Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:51:17.968423828Z caller=silence.go:262 component=silences msg="Running maintenance"
level=debug ts=2023-07-31T03:51:17.968720816Z caller=nflog.go:334 component=nflog msg="Running maintenance"
level=debug ts=2023-07-31T03:51:17.969240548Z caller=silence.go:264 component=silences msg="Maintenance done" duration=825.122µs size=0
level=debug ts=2023-07-31T03:51:17.970623569Z caller=nflog.go:336 component=nflog msg="Maintenance done" duration=1.90442ms size=231
level=debug ts=2023-07-31T03:51:25.678158412Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:51:40.687655305Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:51:55.692023595Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:52:10.686390826Z caller=dispatch.go:343 component=dispatcher aggrGroup="{}:{alertname=\"proxy server down\"}" msg=flushing alerts="[proxy server down[fc7e0b6][active]]"
level=debug ts=2023-07-31T03:52:10.705708192Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:52:25.718514532Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:52:40.727456017Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:52:55.686997853Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:53:10.720848081Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:53:25.742619121Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:53:40.67590946Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:53:55.6884547Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:54:10.692246201Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:54:25.749287246Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:54:40.676786371Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:54:55.681933968Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:55:10.68143208Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:55:25.680483375Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:55:40.705210812Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:55:55.684645606Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:56:10.686986752Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:56:25.699854178Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:56:40.705357826Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:56:55.690212524Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:57:10.680905019Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:57:10.68777953Z caller=dispatch.go:343 component=dispatcher aggrGroup="{}:{alertname=\"proxy server down\"}" msg=flushing alerts="[proxy server down[fc7e0b6][active]]"
level=debug ts=2023-07-31T03:57:25.690059783Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:57:40.689509075Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:57:55.708505116Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:58:10.689716019Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:58:25.718187061Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:58:40.697977025Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:58:55.700138704Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:59:10.68578247Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:59:25.67937207Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:59:40.678370792Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T03:59:55.699916474Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T04:00:10.718151727Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][active]"
level=debug ts=2023-07-31T04:00:25.687306435Z caller=dispatch.go:104 component=dispatcher msg="Received alert" alert="proxy server down[fc7e0b6][resolved]"
# (this is not in logs, but just a bunch of "Running maintenance" logs after this)
```

Basically, the issue can be summarized by the alertmanager not properly flushing this alerts resolved alert. All of the other alerts that I have don't have this issue, and properly send the "resolved" Slack message after sending the "firing" message (after the alert is resolved). No silences involved either. 

As an additional note, sometimes this specific alert works as intended, and sends both "firing" and "resolved" messages. Just wondering why it doesn't flush the "resolved" alert in some cases.

**Comment by benw10-1** [comment]
> You are running **ancient** Prometheus and Alertmanager versions.
> 
> Have you tried to reproduce the problem with current supported versions?

Yeah going to try updating and seeing if the issue persists, thanks for the feedback.

**Comment by benw10-1** [comment]
I seems that updating the version has fixed the issue as far as I can tell. Will re-open if it happens again.
